plugins/plugin-docker/che-plugin-docker-client/src/main/java/org/eclipse/che/plugin/docker/client/DockerConnector.java

```diff
@@ -185,7 +189,48 @@ public Version getVersion() throws IOException {
             if (OK.getStatusCode() != response.getStatus()) {
                 throw getDockerException(response);
             }
-            return parseResponseStreamAsListAndClose(response.getInputStream());
+            return parseResponseStreamAsListAndClose(response.getInputStream(), new TypeToken<List<Image>>() {}.getType());
+        } catch (JsonParseException e) {
+            throw new IOException(e.getLocalizedMessage(), e);
+        }
+    }
+
+    /**
+     * Method returns list of docker containers, include non-running ones.
+     *
+     * @throws IOException
+     *         when problems occurs with docker api calls
+     */
+    public List<ContainerListEntry> listContainers() throws IOException {
+        return listContainers(ListContainersParams.from().withAll(true));
+    }
+
+    /**
+     * Method returns list of docker containers which was filtered by {@link ListContainersParams}
+     *
+     * @throws IOException
+     *         when problems occurs with docker api calls
+     */
+    public List<ContainerListEntry> listContainers(ListContainersParams params) throws IOException {
+        final Filters filters = params.getFilters();
+
+        try (DockerConnection connection = connectionFactory.openConnection(dockerDaemonUri)
+                                                            .method("GET")
+                                                            .path("/containers/json")) {
+            addQueryParamIfNotNull(connection, "all", params.isAll());
+            addQueryParamIfNotNull(connection, "size", params.isSize());
+            addQueryParamIfNotNull(connection, "limit", params.getLimit());
+            addQueryParamIfNotNull(connection, "since", params.getSince());
+            addQueryParamIfNotNull(connection, "before", params.getBefore());
+            if (filters != null) {
+                connection.query("filters", urlPathSegmentEscaper().escape(JsonHelper.toJson(params.getFilters())));
```

Comment:
Either use local variable you've previously set or don't create that variable at all